File: mau/53273.py
def ucln(a, b):
    while b != 0:
        r = a % b
        a = b
        b = r
    return a
    
a, b, c = [int(x) for x in input().split()]
d = ucln(a,b)
e = ucln(c,d)
print(e)

File: mau/53273.cpp
#include <iostream>
using namespace std;
int ucln(int m, int n) {
    int r;
    while (n!=0) {
        r = m % n;
        m = n;
        n = r;
    }
    return m;
}
int main()
{
    int a, b, c;
    cin >> a >> b >> c;
    int d = ucln(a, b);
    int e = ucln(c, d);
    cout << e;
    return 0;
}
